tests/ModelTestCase.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+namespace Tests;
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+use Illuminate\Database\Eloquent\Relations\BelongsToMany;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+
+abstract class ModelTestCase extends TestCase
+{
+    protected function runConfigurationAssertions(Model $model, $assertions)
+    {
+        $assertions = array_merge([
+            'fillable' => [],
+            'hidden' => [],
+            'guarded' => ['*'],
+            'visible' => [],
+            'table' => null,
+            'primaryKey' => 'id',
+            'casts' => ['id' => 'int'],
+            'dates' => ['created_at', 'updated_at'],
+            'collectionClass' => Collection::class,
+            'connection' => null,
+        ], $assertions);
+
+        extract($assertions);
+        $this->assertEquals($assertions['fillable'], $model->getFillable());
+        $this->assertEquals($assertions['guarded'], $model->getGuarded());
+        $this->assertEquals($assertions['hidden'], $model->getHidden());
+        $this->assertEquals($assertions['visible'], $model->getVisible());
+        $this->assertEquals($assertions['casts'], $model->getCasts());
+        $this->assertEquals($assertions['dates'], $model->getDates());
+        $this->assertEquals($assertions['primaryKey'], $model->getKeyName());
+        $c = $model->newCollection();
+        $this->assertEquals($assertions['collectionClass'], get_class($c));
+        $this->assertInstanceOf(Collection::class, $c);
+        if ($assertions['connection'] !== null) {
+            $this->assertEquals($assertions['connection'], $model->getConnectionName());
+        }
+        if ($assertions['table'] !== null) {
+            $this->assertEquals($assertions['table'], $model->getTable());
+        }
+    }
```

Comment:
giải thích function này giúp m với? function này test cái gì?

Reply:
func này test các thuộc tính của model như là fillable, guard, hidden, visible,...

Reply:
cái function to ntn sao có 1 dòng trả lời thế bn ơi @_@ có tâm cho m chút

Reply:
Đầu tiên em tạo ra các mảng chứa các thuộc tính cần test cho model, sau đó so sánh các giá trị trong mảng assertion với các thuộc tính của model bằng hàm assertEquals()
Hàm assertInstanceOf() sẽ thực hiện xác định kết quả so sánh có đúng intance mong muốn hay không
Nếu connection hoặc table khác null thì mới so sánh với thuộc tính connection, table của model